src/plugin.rs

```diff
@@ -1,26 +1,17 @@
+use std::any::{type_name, Any};
+
 use crate::AppBuilder;
-// use libloading::{Library, Symbol};
-use std::any::Any;
 
 /// A collection of Bevy App logic and configuration
 ///
 /// Plugins use [AppBuilder] to configure an [App](crate::App). When an [App](crate::App) registers a plugin, the plugin's [Plugin::build] function is run.
 pub trait Plugin: Any + Send + Sync {
     fn build(&self, app: &mut AppBuilder);
     fn name(&self) -> &str {
-        std::any::type_name::<Self>()
+        type_name::<Self>()
+    }
+    /// If you override this to return true, then you will be allowed to add your plugin multiple times.
+    fn can_add_multiple_times(&self) -> bool {
+        false
     }
 }
```

Comment:
If we decide to create a macro to build an impl for this, here would be a good place to put it.

Reply:
Hmmm... I feel like there's some niceties we could apply using a DSL of some kind, but the impls are pretty lightweight overall IMO.